### PR TITLE
Feature/dark mode

### PR DESCRIPTION
**Pull request by rabin245**
Added theme switcher button in the navbar of the application.

### CODE REVIEW

File: src/components/ThemeSwitcher/index.jsx [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import SunIcon from "./SunIcon";
+import MoonIcon from "./MoonIcon";
+import { useContext } from "react";
+import { ThemeContext } from "../../context/ThemeContext";
+import styles from "./ThemeSwitcher.module.css";
+
+const ThemeSwitcher = () => {
+  const { darkMode, switchTheme } = useContext(ThemeContext);
```
Comment (sagarkhanal07) [inline src/components/ThemeSwitcher/index.jsx:8]: Remove app logic from component. Update it to props

File: src/components/ThemeSwitcher/SunIcon.jsx
Comment (sagarkhanal07) [inline src/components/ThemeSwitcher/SunIcon.jsx:1]: Is this asset file? Maybe you can move it to it's respective folder

File: src/components/Menu/index.jsx [hunk 2]
```diff
@@ -9,6 +10,7 @@ const Menu = ({ homeRef, aboutRef, contactRef, skillsRef, projectsRef }) => {
       <MenuItem title="Skills" reference={skillsRef} />
```
Comment (sagarkhanal07) [inline src/components/Menu/index.jsx:10]: Try to remove hardcode titles from component. Make it as reusable as possible

**Review by sagarkhanal07 — CHANGES_REQUESTED** [review]

File: src/components/Menu/index.jsx (hunk 2 above)
Comment (rabin245) [reply src/components/Menu/index.jsx:10]: Should I make an array of { title, ref } objects in the component and map them to create a list of <MenuItem/> components?

File: src/components/Menu/index.jsx [hunk 3]
```diff
@@ -1,14 +1,31 @@
+import { useContext } from "react";
 import MenuItem from "../MenuItem";
+import ThemeSwitcher from "../ThemeSwitcher";
 import styles from "./Menu.module.css";
+import { ThemeContext } from "../../context/ThemeContext";
 
 const Menu = ({ homeRef, aboutRef, contactRef, skillsRef, projectsRef }) => {
+  const { darkMode, switchTheme } = useContext(ThemeContext);
+
+  const menuItems = [
```
Comment (sagarkhanal07) [inline src/components/Menu/index.jsx:10]: Make this menuItems to passable as props

File: src/components/Menu/index.jsx [hunk 4]
```diff
@@ -1,14 +1,31 @@
+import { useContext } from "react";
 import MenuItem from "../MenuItem";
+import ThemeSwitcher from "../ThemeSwitcher";
 import styles from "./Menu.module.css";
+import { ThemeContext } from "../../context/ThemeContext";
 
 const Menu = ({ homeRef, aboutRef, contactRef, skillsRef, projectsRef }) => {
+  const { darkMode, switchTheme } = useContext(ThemeContext);
```
Comment (sagarkhanal07) [inline src/components/Menu/index.jsx:8]: avoid using context/logic inside components.

**Review by sagarkhanal07 — CHANGES_REQUESTED** [review]
Update few changes as commented

File: src/components/Menu/index.jsx (hunk 3 above)
Comment (rabin245) [reply src/components/Menu/index.jsx:10]: Passed the menuItems as props from Header component.

**Review by aayussraut — APPROVED** [review]

**Pull request merged**